shaolinmkz/fast-food-fast
# #160535014 Refactor controllers and Middlewares codes to be clean and DRY, make the food menu routes work

#### What does this PR do?
Have `code climate` maintainability grade increase and make menus route functional

#### Description of Task to be completed?
Having the following task completed:
* Abstract controller codes
* Eliminate duplication of function (similar codes)
* Reduce excessive if and nested if and nested for loop statement
* Make sure codes doesn't exceed 100 lines
* Minimized the code length of functions and methods from exceeding 25 lines
* Added menu routes
* Tested the menu routes
* Update `.readme` file to standard
* Fix all wrong indentations
* Change all codes to ES6 and above (functions and variable declaration etc)
* Add a line at the end of each code block
* Document functions, classes and methods using `jsdoc`
* Improved code base with a `coverage` of 100% throughout

#### How should this be manually tested?
After cloning the repo, cd into it and RUN `npm test` to see the test result and coverage.
After that run `npm start` 
* Using Postman test every endpoint ablove with this header:
_key_:  Content-Type value: application/json

#### Any background context you want to provide?
There is an implementation that is yet to be completed which is using the store price in the data structure to generate bills for the customer. I am still generating random values for bills.
 
#### What are the relevant pivotal tracker stories?
#160535014
